src/main.py

```diff
@@ -40,16 +40,25 @@ def main():
     habana_dedicated = os.getenv("HABANA_DEDICATED", "false").lower() == "true"
     hf_trcomp_dedicated = os.getenv("HUGGINFACE_TRCOMP_DEDICATED", "false").lower() == "true"
     trcomp_dedicated = os.getenv("TRCOMP_DEDICATED", "false").lower() == "true"
+    image_type = os.getenv("IMAGE_TYPE", "").lower()
+    training_dedicated = image_type == "training"
+    inference_dedicated = image_type == "inference"
 
     # Get config value options
-    frameworks_to_skip = parse_dlc_developer_configs("build", "skip_frameworks")
+    frameworks_to_build = parse_dlc_developer_configs("build", "build_frameworks")
+    training_enabled = parse_dlc_developer_configs("build", "build_training")
+    inference_enabled = parse_dlc_developer_configs("build", "build_inference")
     ei_build_mode = parse_dlc_developer_configs("dev", "ei_mode")
     neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
     graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
     habana_build_mode = parse_dlc_developer_configs("dev", "habana_mode")
     trcomp_build_mode = parse_dlc_developer_configs("dev", "trcomp_mode")
     hf_trcomp_build_mode = parse_dlc_developer_configs("dev", "huggingface_trcomp_mode")
 
+    # Condition to check whether we are training or inference dedicated/enabled
+
+    train_or_inf_enabled = (training_dedicated and training_enabled) or (inference_dedicated and inference_enabled)
```

Comment:
may want to add condition to check for "" to still support older types of PR jobs if we want to bring them back in any capacity